docs/proposals/20221003-In-place-propagation-of-Kubernetes-objects-only-changes.md

```diff
@@ -0,0 +1,321 @@
+
+As a consequence we have decided to expand this work to consider how to propagate labels, annotations and fields impacting only Kubernetes objects or
+controller behaviour, as well as this related issue: [Labels and annotations for MachineDeployments and KubeadmControlPlane created by topology controller](https://github.com/kubernetes-sigs/cluster-api/issues/7006).
+
+### Goals
+
+- Define how labels and annotations propagate from ClusterClass to KubeadmControlPlane/MachineDeployments and ultimately to Machines.
+- Define how fields impacting only Kubernetes objects or controller behaviour propagate from ClusterClass to KubeadmControlPlane
+  MachineDeployments, and ultimately to Machines.
+- Define how to prevent that propagation of labels, annotations and other fields impacting only Kubernetes objects or controller behaviour
+  trigger unnecessary rollouts.
+
+### Non-Goals
+
+- Discuss the immutability core design principle in Cluster API (on the contrary, this proposal makes immutability even better by improving
+  the criteria on when we trigger Machine rollouts).
+- To support in-place mutation for components or settings that exist on Machines (this proposal focuses only on labels, annotations and other
+  fields impacting only Kubernetes objects or controller behaviour).
+
+### Future-Goals
+
+- Expand propagation rules including MachinePools after the [MachinePools Machine proposal](./20220209-machinepool-machines.md) is implemented.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin/user, I would like a declarative and secure means by which to assign roles to my nodes via Cluster topology metadata
+(for Clusters with ClusterClass).
+
+As a cluster admin/user, I would like a declarative and secure means by which to assign roles to my nodes via KubeadmControlPlane and
+MachineDeployments (for Clusters without ClusterClass).
+
+#### Story 2
+
+As a cluster admin/user, I would like to change labels or annotations on Machines without triggering Machine rollouts.
+
+#### Story 3
+
+As a cluster admin/user, I would like to change nodeDrainTimeout on Machines without triggering Machine rollouts.
+
+#### Story 4
+
+As a cluster admin/user, I would like to set autoscaler labels for MachineDeployments by changing Cluster topology metadata
+(for Clusters with ClusterClass).
+
+### Implementation Details/Notes/Constraints
+
+### Metadata propagation
+
+The following schema represent how metadata propagation works today (also documented in [book](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/metadata-propagation.html)).
+
+![Figure 1](./images/in-place-propagation/current-state.png)
+
+With this proposal we are suggesting to improve metadata propagation as described in the following schema:
+
+![Figure 2](./images/in-place-propagation/proposed-changes.png)
```

Comment:
* The line from InfraMachineTemplate to BootstrapConfig shouldn't exist

Reply:
Q: Is there a way to modify those png's going forward? If yes, how? 
(e.g. when we want to include them in our book and then have to modify them over time)

Reply:
Fixed images, added original pptx